.pre-commit-config.yaml

```diff
@@ -0,0 +1,33 @@
+exclude: "\
+    ^(\
+    .changes|\
+    .github|\
+    awscli/examples|\
+    awscli/topics|\
+    awscli/botocore|\
+    awscli/s3transfer|\
+    awscli/doc|\
+    exe/assets|\
+    tests/functional/cloudformation/deploy_templates/booleans/input.yml|\
+    tests/functional/cloudformation/deploy_templates/nested-tag/input.yml|\
+    tests/|\
+    CHANGELOG.rst|\
+    configure\
+    )"
+repos:
+  - repo: 'https://github.com/pre-commit/pre-commit-hooks'
+    rev: v4.5.0
+    hooks:
+      - id: check-yaml
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+  - repo: 'https://github.com/PyCQA/isort'
+    rev: 5.12.0
+    hooks:
+      - id: isort
```

Comment:
I don't think we actually need this for CLI v2. It's included for backwards compat in botocore imports. IMO we should just delete those here.

Reply:
I'm not sure I understand what you're suggesting here. `isort` makes changes in the CLI v2 codebase.

Reply:
`ruff` has the same functionality as `isort` built in, you just add the `I` flag in the config.

The only reason we still have isort in botocore is because ruff does cleanup on all unused imports. There should be a full explanation in the original PR. Our ideal end state doesn't use `isort` at all.

Reply:
Original PR: https://github.com/boto/botocore/pull/3206#issue-2353772183

Reply:
Ah, I saw the part about quote configuration but not the part about `isort`. I don't have `I` right now but I can make that change. Thanks!

Reply:
> ruff does cleanup on all unused imports

This caused a lot of issues for the CLI codebase too, so I am disabling that with `ruff` until it can be fixed separately:

https://github.com/aws/aws-cli/pull/9123/files#diff-50c86b7ed8ac2cf95bd48334961bf0530cdc77b5a56f852c5c61b89d735fd711R185

```
ignore = ["F401"]
```

Reply:
I dropped `isort` from the precommit config and removed the configuration section from `pyproject.toml`, and added the `I` flag to the ruff configuration section.

Reply:
Done, resolving.